exposed-java-time/src/main/kotlin/org/jetbrains/exposed/sql/javatime/JavaDateFunctions.kt

```diff
@@ -49,7 +49,7 @@ object CurrentDate : Function<LocalDate>(JavaLocalDateColumnType.INSTANCE) {
     }
 }
 
-class CurrentTimestamp<T : Temporal> : Expression<T>() {
+class CurrentTimestamp<T : Temporal> : Function<T>(JavaInstantColumnType.INSTANCE) {
```

Comment:
As mentioned in the PR discussion, there may be edge cases not exhausted by the current test suites that arise from restricting the function column type.

`CurrentTimestamp()` doesn't have a way of knowing the type of the column it's being used with, unless the user provides it directly, so the following is an alternative if there are concerns:
```kt
class CurrentTimestamp<T : Temporal>(
    columnType: IColumnType = JavaInstantColumnType.INSTANCE
) : Function<T>(columnType) { }
```

The column type chosen is possibly more a concern for functions that are commonly stand-alone and can be used to create their own column of generated values. If this is being used a default function that returns a timestamp literal, retrieving DB values will be dependent on the type of the column on which `defaultExpression()` is called.